src/Tribe/Views/V2/View.php

```diff
@@ -1049,4 +1050,58 @@ public function get_today_url( $canonical = false ) {
 
 		return Rewrite::instance()->get_canonical_url( $ugly_url );
 	}
+
+	/**
+	 * Builds the link label to use from the URL.
+	 *
+	 * This is usually used to build the next and prev link URLs labels.
+	 * Extending classes can customize the format of the the label by overriding the `get_label_format` method.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $url The input URL to build the link label from.
+	 *
+	 * @return string The formatted and localized, but not HTML escaped, link label.
+	 *
+	 * @see View::get_label_format(), the method child classes should override to customize the link label format.
+	 */
+	public function get_link_label( $url ) {
+		if ( empty( $url ) ) {
+			return '';
+		}
+
+		$url_query = parse_url( $url, PHP_URL_QUERY );
+
+		if ( empty( $url_query ) ) {
+			return '';
+		}
+
+		parse_str( $url_query, $args );
+
+		$date = Arr::get_first_set( $args, [ 'eventDate', 'tribe-bar-date' ], false );
+
+		if ( false === $date ) {
+			return '';
+		}
+
+		$date_object = Dates::build_date_object( $date );
+
+		return date_i18n( $this->get_label_format(), $date_object->getTimestamp() + $date_object->getOffset() );
+	}
+
+	/**
+	 * Returns the date format, a valid PHP `date` function format, that should be used to build link labels.
+	 *
+	 * This format will, usually, apply to next and previous links.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The date format, a valid PHP `date` function format, that should be used to build link labels.
+	 *
+	 * @see View::get_link_label(), the method using this method to build a link label.
+	 * @see date_i18n() as the formatted date will, then, be localized using this method.
+	 */
+	protected function get_label_format() {
+		return 'Y-m-d';
```

Comment:
Maybe we want to add a filter here? Or at least leave a `@todo` to discuss it.

Reply:
Child classes are supposed to override this method.
We can add the filter for third-parties, but our code should override the method, not filter it.

Reply:
I've moved the filtering, per-view, in the `View::get_link_label` method.